subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c

```diff
@@ -319,9 +324,26 @@ struct pdu_adv *lll_adv_pdu_alloc_pdu_adv(void)
 	p = MFIFO_DEQUEUE(pdu_free);
 	LL_ASSERT(p);
 
+#if defined(CONFIG_BT_CTLR_ADV_PDU_LINK)
+	PDU_ADV_NEXT_PTR(p) = NULL;
+#endif
 	return p;
 }
 
+#if defined(CONFIG_BT_CTLR_ADV_PDU_LINK)
+void lll_adv_pdu_linked_release_from_ull(struct pdu_adv *pdu)
```

Comment:
`from_ull` means this should be called from ULL only
I can change this name to 'release_all' and/or put a comment in header to explain, I do not really think overly verbose function names are good idea since they are just too long, there has to be some compromise made.

Reply:
@cvinayak what do you think about such (`*_from_ull`) convention? If that goes into master then we will establish a scheme that other contributors may follow.